src/Platform/Microsoft.Testing.Platform/Hosts/TestHostBuilder.cs

```diff
@@ -283,10 +286,76 @@ internal class TestHostBuilder(IFileSystem fileSystem, IRuntimeFeature runtimeFe
             return toolsTestHost;
         }
 
+        NamedPipeClient? namedPipeClient = null;
+
+        // If we are in server mode and we the pipe name is provided
+        // then, we need to connect to the pipe server.
+        if (commandLineHandler.TryGetOptionArgumentList(PlatformCommandLineProvider.ServerOptionKey, out string[]? serverArgs) &&
+            serverArgs?.Length > 0 &&
+            serverArgs[0].Equals(ServerOptionValue, StringComparison.Ordinal) &&
+            commandLineHandler.TryGetOptionArgumentList(PlatformCommandLineProvider.DotNetTestPipeOptionKey, out string[]? arguments))
+        {
+            namedPipeClient = new(arguments[0]);
+
+            namedPipeClient.RegisterSerializer<CommandLineOptionMessages>(new CommandLineOptionMessagesSerializer());
+            namedPipeClient.RegisterSerializer<VoidResponse>(new VoidResponseSerializer());
+
+            await namedPipeClient.ConnectAsync(testApplicationCancellationTokenSource.CancellationToken);
+        }
+
         // If --help is invoked we return
         if (commandLineHandler.IsHelpInvoked())
         {
-            await commandLineHandler.PrintHelpAsync(toolsInformation.Tools);
+            if (namedPipeClient is not null)
+            {
+                List<CommandLineOptionMessage> commandLineHelpOptions = new();
+                foreach (ICommandLineOptionsProvider commandLineOptionProvider in commandLineHandler.CommandLineOptionsProviders)
+                {
+                    foreach (CommandLineOption commandLineOption in commandLineOptionProvider.GetCommandLineOptions())
+                    {
+                        string arity = string.Empty;
+
+                        if (commandLineOption.Arity == ArgumentArity.Zero)
+                        {
+                            arity = "Zero";
+                        }
+
+                        if (commandLineOption.Arity == ArgumentArity.ZeroOrOne)
+                        {
+                            arity = "ZeroOrOne";
+                        }
+
+                        if (commandLineOption.Arity == ArgumentArity.ZeroOrMore)
+                        {
+                            arity = "ZeroOrMore";
+                        }
+
+                        if (commandLineOption.Arity == ArgumentArity.OneOrMore)
+                        {
+                            arity = "OneOrMore";
+                        }
+
+                        if (commandLineOption.Arity == ArgumentArity.ExactlyOne)
+                        {
+                            arity = "ExactlyOne";
+                        }
+
+                        commandLineHelpOptions.Add(new CommandLineOptionMessage(
+                            commandLineOption.Name,
+                            commandLineOption.Description,
+                            arity,
```

Comment:
The values seem the same why not commandLineOption.Arity.ToString() ? Or am I overlooking something?

Reply:
Because we can't have this value from arity property as we have:
public static readonly ArgumentArity ZeroOrMore = new(0, int.MaxValue);